ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -119,6 +119,15 @@ boolean addLocalTransaction(final Transaction transaction) {
     return addTransaction;
   }
 
+  void evictOldTransactions(Instant maximumRetentionPeriod) {
+    for (final TransactionInfo transactionInfo : prioritizedTransactions) {
+      int difference = transactionInfo.getAddedToPoolAt().compareTo(maximumRetentionPeriod);
+      if (difference > 0) {
+        pendingTransactions.remove(transactionInfo.getTransaction().hash());
```

Comment:
This isn't fully removing the transaction from the pool (prioritizedTransactions and transactionsBySender aren't updated and dropped transaction notifications aren't sent).  It needs to call `removeTransaction` to ensure the removal is done correctly.

Once that's fixed, I suspect this will cause a `ConcurrentModificationException` when there are multiple transactions in the pool because you're removing from the set as you iterate it.  You'll need to create a list of transactions to remove with something like:
```
final List<TransactionInfo> transactionsToRemove = prioritizedTransactions.stream().filter(transaction -> transaction.getAddedToPoolAt().isBefore(earliestAllowedAddTime)).collect(toList());
transactionsToRemove.forEach(this::removeTransaction)
```